packages/babel-helpers/src/helpers.js

```diff
@@ -2015,6 +2015,24 @@ helpers.classPrivateMethodGet = helper("7.1.6")`
   }
 `;
 
+helpers.classPrivateFieldInitSpec = helper("7.14.1")`
+  export default function _classPrivateFieldInitSpec(obj, privateMap, value) {
+    if (privateMap.has(obj)) {
+      throw new TypeError("redeclared private field identifier");
+    }
+    privateMap.set(obj, value);
+  }
+`;
+
+helpers.classPrivateMethodInitSpec = helper("7.14.1")`
+  export default function _classPrivateMethodInitSpec(obj, privateSet) {
+    if (privateSet.has(obj)) {
+      throw new TypeError("redeclared private method identifier");
+    }
```

Comment:
We can extract this code in a third helper and share it between the two helpers:

```js
import checkPrivateRedeclaration from "checkPrivateRedeclaration";

export default function _classPrivateMethodInitSpec(obj, privateSet) {
  checkPrivateRedeclaration(privateSet, obj);
  privateSet.add(obj);
}
```